### PR TITLE
Fetches 1pass secrets in task

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/12723

This PR mostly just establishes a pattern to access 1Password secrets through [XComs](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html).  It is meant to avoid making many unnecessary API calls to the 1pass connect server, as discussed in [this slack thread](https://austininnovation.slack.com/archives/CHZE6BC6L/p1687530254736569)

I got carried away with README edits! I'm happy to revise/revert: feedback is very welcome.

## Associated repo
- atd-airflow

## Testing

**Steps to test:**
1. Checkout this branch and spin up Airflow as usual:
```
$ docker compose up
```
2. Trigger any of our existing service bot DAGs: they should run normally

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [ ] Add note to 1PW secrets moved to API vault and check for duplicates

### CODE REVIEW

File: dags/atd_service_bot_issues_to_socrata.py [hunk 1]
```diff
@@ -62,16 +63,26 @@
...
+    render_template_as_native_obj=True,
 ) as dag:
+    @task(
+        task_id="get_env_vars",
+        execution_timeout=timedelta(seconds=30),
+    )
+    def get_env_vars():
+        from utils.onepassword import load_dict
+        env_vars = load_dict(REQUIRED_SECRETS)
+        return env_vars
```
Comment (johnclary) [inline dags/atd_service_bot_issues_to_socrata.py:74]: by default, anything returned from a python `task` will be set into the XComs key called `return_value`.

File: dags/atd_service_bot_issues_to_socrata.py [hunk 2]
```diff
@@ -62,16 +63,26 @@
     dagrun_timeout=timedelta(minutes=60),
     tags=["repo:atd-service-bot", "socrata", "github"],
     catchup=False,
+    render_template_as_native_obj=True,
```
Comment (johnclary) [inline dags/atd_service_bot_issues_to_socrata.py]: this was the trick to getting this working: it allows us to access a `dict` from within a template variable without having to deal with string parsing ✨

File: dags/atd_service_bot_issues_to_socrata.py [hunk 3]
```diff
@@ -62,16 +63,26 @@
...
+        return env_vars
+
     t1 = DockerOperator(
         task_id="dts_github_to_socrata",
         image=docker_image,
         api_version="auto",
         auto_remove=True,
         command="./atd-service-bot/issues_to_socrata.py",
-        environment=env_vars,
+        environment="{{ task_instance.xcom_pull(task_ids='get_env_vars') }}",
```
Comment (johnclary) [inline dags/atd_service_bot_issues_to_socrata.py]: by default the `task_instance.xcom_pull()` method is going to pull from the `return_value` key. so this template variables pipes our secrets dictionary right into the Docker environment.

File: README.md [hunk 4]
```diff
@@ -1,18 +1,36 @@
...
+  - [Tags](#tags)
+  - [Moving to production](#moving-to-production)
+- [Utilities](#utilities)
+  - [1Password utility](#1password-utility)
+  - [Slack operator utility](#slack-operator-utility)
+- [Useful Commands](#useful-commands)
+- [Updating the stack](#updating-the-stack)
+- [HAProxy and SSL](#haproxy-and-ssl)
+  - [HAProxy operation](#haproxy-operation)
+
```
Comment (frankhereford) [inline README.md:28]: ❤️ 🚀 💯

**Review by frankhereford — APPROVED** [review]
@johnclary - this is two high quality additions to the stack!!!


#### ✅ Working out the `xcoms` solution is no small task. This is 🚀

🤔 I am wondering, does Airflow 2's inclusion of the "[TaskFlow](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html)" system make this easier? Here's how I was passing simple python data around in my old test DAG, and it felt pythonic, if that's the right use of that word. Here's where [I was doing that](https://github.com/frankhereford/airflow/blob/main/dags/weather.py#L138-L143). I feel like I may be missing something subtle though in your usage, and even with that maybe being true, I am 100% on board with this improvement to our DAG pattern. This is great.

#### 📖 The changes to the documentation are awesome

This reads so well, and you've got it organized and accessible. The TOC takes the cake. This looks like an extra polished stack with that README!

File: README.md (hunk 4 above)
Comment (mddilley) [reply README.md:28]: +1 - love the polish and the new additions about the new xcoms addition. 🙏

File: README.md [hunk 5]
```diff
@@ -27,50 +45,120 @@ OP_CONNECT=<Get from 1Password entry named "Endpoint for 1Password Connect Serve
...
+    get_env_vars() >> my_docker_task
 ```
-docker compose run --rm airflow-cli dags test <dag-id>
+
+### Slack operator utility
+
+The Slack operator utility makes use of the integration between the Airflow and a Slack app webhook. The purpose of the utility is to add Slack notifications to DAGs using the [callback](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/callbacks.html#callback-types) parameters. Failure, critical failure, and success notifications are implemented.
+
+To configure the Slack operator in your local instance, from the Airflow UI go to **Admin** > **Connections** and choose **Slack API** as the **connection type**. You can find the remaining settings in 1Password under the **Airflow - Slack Bot** item.
```
Comment (mddilley) [inline README.md:137]: 🙏

**Review by mddilley — APPROVED** [review]
This looks awesome, John! Thanks for adding some of the finer details of what to expect with local dev and what it looks like to move something to prod to the readme. Xcoms is a TIL for me, and I'm glad you found a workaround for what was starting to feel more and more like a loose end. 🚀 

I tested the service bot intake script locally, and it found no new service requests successfully. ✅

File: README.md (hunk 4 above)
Comment (johnclary) [reply README.md:28]: credit where it's due, i have an extension that makes the TOC: https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one

**Comment by johnclary** [comment]
> does Airflow 2's inclusion of the "[TaskFlow](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html)" system make this easier?

@frankhereford - lol! i totally missed that you can just return values from tasks now 🤦  i'm going to give this a try locally and push it up if it works: so much cleaner 🙏

**Comment by johnclary** [comment]
ok, here's another iteration using the simple and intuitive taskflow pattern: thanks again @frankhereford!

**Review by frankhereford — APPROVED** [review]
This looks so good, John. Readable and tidy - awesome patterns, and the docs are just great. Out of the park! 🏟️⚾️

**Comment by johnclary** [comment]
thanks @frankhereford will merge/deploy on Monday

**Review by mddilley — APPROVED** [review]
Tested the service bot intake locally and this still works perfectly! I need to check out the new Airflow 2 TaskFlow capabilities. Thanks for the TIL, @frankhereford! 🙏

**Pull request merged**